File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{

  string text;
  int letters, words, sentences, index;
  double L, S;

  text = get_string("Text: ");
  letters = count_letters(text);
  words = count_words(text);
  sentences = count_sentences(text);

  //Average number of letters per 100 words
  L = letters / (float)words * 100;
  //Average number of sentences per 100 words
  S = sentences / (float)words * 100;
  //Coleman-Liau formula
  index = round(0.0588 * L - 0.296 * S - 15.8);

  //print the grade level as defined by the Coleman-Liau index (e.g. "Grade 2" or "Grade 8" or the like).
  if (index > 16)
  {
    printf("Grade 16+\n");
  }
  else if (index < 1)
  {
    printf("Before Grade 1\n");
  }
  else
  {
    printf("Grade %i\n", index);
  }

}

int count_letters(string text)
{
  int length, counter = 0;
  //store the text length
  length = strlen(text);

  for (int i = 0; i < length; i++)
  {
    if (isalpha(text[i]) != 0)
    {
      counter++;
    }

  }

  return counter;

}

int count_words(string text)
{
  int length, counter = 0;
  //store the text length
  length = strlen(text);

  for (int i = 0; i < length; i++)
  {
    if (isblank(text[i]))
    {
      counter++;
    }

  }
  counter++;

  return counter;

}

int count_sentences(string text)
{
  int length, counter = 0;
  //store the text length
  length = strlen(text);

  for (int i = 0; i < length; i++)
  {
    if (text[i] == '.' || text[i] == '?' || text[i] == '?')
    {
      counter++;
    }

  }

  return counter;

}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calculate the average number of letters and sentences per 100 words
    L = letters / words * 100
    S = sentences / words * 100

    # Calculate the Coleman-Liau index
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Print the grade level as defined by the Coleman-Liau index (e.g. "Grade 2" or "Grade 8" or the like)
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def count_letters(text):
    # Initialize the counter variable
    counter = 0

    # Loop through each character in the text
    for char in text:
        # Check if the character is an alphabetic letter
        if char.isalpha():
            counter += 1

    # Return the count of letters
    return counter


def count_words(text):
    # Split the text into words based on whitespace characters
    words = text.split()

    # Return the count of words
    return len(words)


def count_sentences(text):
    # Initialize the counter variable
    counter = 0

    # Loop through each character in the text
    for char in text:
        # Check if the character is a period, exclamation mark, or question mark
        if char in [".", "!", "?"]:
            counter += 1

    # Return the count of sentences
    return counter


main()
